creator/lib/screens/login_screen.dart

```diff
@@ -8,10 +17,221 @@ class LoginScreen extends StatefulWidget {
+
+  //TODO: _addToDatabaseでデータベースに登録するが、現時点では実装していない。
+  void _navigateBasedOnRole() {
+    final _auth = FirebaseAuth.instance.currentUser;
+    print('宏輝: $_auth');
+    FirebaseFirestore.instance
+        .collection('users')
+        .doc(_auth?.email)
+        .get()
+        .then((DocumentSnapshot snapshot) {
+      if (snapshot.get(FieldPath(['role'])) == 0) {
+        Navigator.of(context).pushReplacementNamed('/salon_applying_screen');
+      } else if (snapshot.get(FieldPath(['role'])) == 1) {
+        FirebaseFirestore.instance
+            .collection('salons')
+            .where('owner', isEqualTo: _auth?.email)
+            .get()
+            .then((QuerySnapshot snapshot) {
+          print(snapshot.docs.isNotEmpty);
+          print(snapshot.docs.isEmpty);
+          if (snapshot.docs.isNotEmpty) {
+            Navigator.of(context)
+                .pushReplacementNamed(('/salon_applying_screen'));
+          } else {
+            Navigator.of(context)
+                .pushReplacementNamed(('/salon_applying_screen'));
+          }
+        });
+      }
+    });
+  }
+
+  Future<void> _addToDatabase() async {
+    DbHandler dbHandler = DbHandler();
+    final User? user = FirebaseAuth.instance.currentUser;
+
+    // データが無いとエラー可能性
+    if (user != null) {
+      //TODO(hiroki): 強制アンラップのハンドリング
+      if (await dbHandler.getUser(user.email!) != null) {
+        return;
+      }
+      assert(!user.isAnonymous);
+      assert(await user.getIdToken() != null);
+
+      dbHandler
+          .addUser(UserModel(
+            //TODO(hiroki): 強制アンラップのハンドリング
+            email: user.email!,
+            //TODO(hiroki): 強制アンラップのハンドリング
+            profile: UserProfileModel(name: user.displayName!, aboutMeText: ''),
+            role: RoleType.member,
+            settings: UserSettings(pushNotifications: true),
+            created: DateTime.now().toUtc(),
+          ))
+          .onError((error, stackTrace) => _buildLoginFailedDialog());
+    }
+  }
+
+  Future _buildLoginFailedDialog() {
+    return showDialog(
+      context: context,
+      builder: (_) {
+        return AlertDialog(
+          content: Text("ログインに失敗しました\nもう一度お試しください"),
+          actions: [
+            TextButton(
+              onPressed: () {
+                Navigator.pop(context);
+              },
+              child: Text('戻る'),
+            ),
+          ],
+        );
+      },
+    );
+  }
+
+  void _tryLoginWith(BuildContext context, method) async {
+    final User? user = FirebaseAuth.instance.currentUser;
+    setState(() {
+      _loginInProgress = true;
+    });
+    try {
+      await method.whenComplete(
+        () async {
+          await _addToDatabase();
+          setState(
+            () {
+              _loginInProgress = false;
+              userLoggedIn = true;
+            },
+          );
+        },
+      );
+    } on FirebaseAuthException catch (e) {
+      print(e.message);
+    }
```

Comment:
Do _loginInProgress = false in setState at the end, before the final navigation.

Reply:
I moved function.